pkg/collector/py/loader_test.go

```diff
@@ -38,13 +39,76 @@ func TestLoad(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Zero(t, len(instances))
 
+	// the python module works
+	config = integration.Config{Name: "working"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	config.Instances = append(config.Instances, []byte("bar: baz"))
+	instances, err = l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 2, len(instances))
+
 	// the python module is good but nothing derives from AgentCheck
 	config = integration.Config{Name: "foo"}
 	instances, err = l.Load(config)
 	assert.NotNil(t, err)
 	assert.Zero(t, len(instances))
 }
 
+func TestLoadVersion(t *testing.T) {
+	l, _ := NewPythonCheckLoader()
+
+	// the python module has no version
+	config := integration.Config{Name: "working"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	instances, err := l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(instances))
+
+	// the python module has a version
+	config = integration.Config{Name: "version"}
+	config.Instances = append(config.Instances, []byte("foo: bar"))
+	instances, err = l.Load(config)
+	require.Nil(t, err)
+	assert.Equal(t, 1, len(instances))
+	assert.Equal(t, "1.0.0", instances[0].Version())
+}
+
+func TestLoadVersionLock(t *testing.T) {
+	glock := newStickyLock()
```

Comment:
The StickyLock also acquire the GIL which might impact our python code. Could we only lock the OS thread with `runtime.LockOSThread()` here to force python call on the same thread.

Reply:
Okay I see